test/unit/controllers/ChoiceControllerSpec.scala

```diff
@@ -36,14 +39,20 @@ class ChoiceControllerSpec extends ControllerSpec with OptionValues {
   private def existingDeclaration(choice: DeclarationType = DeclarationType.SUPPLEMENTARY) =
     aDeclaration(withId("existingDeclarationId"), withType(choice))
 
-  val choicePage = new choice_page(mainTemplate, minimalAppConfig)
+  val choicePage = mock[choice_page]
 
   val controller =
     new ChoiceController(mockAuthAction, stubMessagesControllerComponents(), choicePage)(ec)
 
+
+  override protected def afterEach(): Unit = {
+    reset(choicePage)
+    super.afterEach()
+  }
```

Comment:
Logic order will be first `beforeEach` after `afterEach`. Of course it doesn't change anything ;)

Reply:
Done! :)